### ISSUE TITLE
Warning module (core) is missing

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1470]
`Warning` module is available in core since 2.4. See https://ruby-doc.org/core-2.4.0/Warning.html

This is used for instance in ROM https://github.com/rom-rb/rom/blob/e444df079a4d903205aea4a60962d0c5ef0447af/changeset/spec/spec_helper.rb#L65

So their specs fail there due to that.

To reproduce

```
Warning
```

Error:
```
NameError: uninitialized constant Warning
	from (irb):1:in `const_missing'
	from (irb):1
	from ~/.rubies/truffleruby-1.0.0-rc9/bin/irb:29:in `<main>'
```

**Comment by chrisseaton** [comment]
We do have specs for this, so it should be easy enough to implement. It may be tricker to route all our existing warnings through it, though, and I don't think that bit is specified.

**Comment by eregon** [comment]
Just a note,
```ruby
warning_api_available = defined?(Warning)
```
Would be a lot better to detect if it's available.

But of course it's good to implement it as well.

**Comment by chrisseaton** [comment]
This is fixed in 1d08927dd34081a478b79564be7b5aab009ac433 and will be in the next release.
